src/mongoose_xmpp_errors.erl

```diff
@@ -12,6 +12,14 @@
 -compile([export_all]).
 -include("mongoose_ns.hrl").
 
+to_error(A) ->
```

Comment:
can we pass a map here?
Something like `#{xmpp_error => conflict}`.
Because I would expect backend modules to add an extra info.
I don't like how we pass stacktrace, for example.
Than it would be easier to extend later and to refactor that stacktrace code, while don't care about breaking this feature much.

Reply:
Use a map here at this place or rather at the handler in `mod_mam`? What other data would you `mongoose_xmpp_error:to_error/1` expect to handle and how? Error reporting/logging?

Reply:
Backend module should return a map to mod_mam.
One of the fields in a map can be used as a reason of error, which can be formatted by this module. We can introduce the map in a separate PR, than the code here is fine.

I don't see a point of passing the whole map here, ErrorType atom would be enough, like it's in the PR.

Reply:
I agree that map is probably not needed here, as this function might be reused by other modules (MUC Light perhaps?)